### ISSUE TITLE
Optional Prop Fields

### ISSUE THREAD

**Issue by jafioti** [sycamore-rs/sycamore#483]
For components with many fields, often many of these are optional. It would be a big ergonomic win to forego these when calling the component. Props can be layed out like so:
```rust
#[derive(Prop)]
struct Props {
    name: String,
    age: i32,
    custom_style: Option<String>, // Optional
}
```
And since custom_style is an Option, if the user doesn't put this in the component call, it will be set to None, otherwise any value passed in will be wrapped in Some.

If we still want to allow users to use normal Options with normal functionality, perhaps this special functionality can be enabled with a flag:
```rust
#[derive(Prop)]
struct Props {
    name: String,
    age: i32,
    #[optional]
    custom_style: Option<String>, // Optional
}
```
Dioxus allows for optional props with their #[inline_props] macro, so look to them for reference.

**Comment by johansmitsnl** [comment]
You can do this:

```rust
#[derive(Prop)]
struct Props {
    name: String,
    age: i32,
    #[builder(default)]
    custom_style: Option<String>, // Optional
}
```

`#[builder(default)]` used the default if not given.

**Comment by jafioti** [comment]
I agree, and if it's by default, then I would think any args passed  in would automatically be wrapped in a Some(), rather than each arg passed requiring the Some() wrapping
